client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -43,14 +45,34 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
     const {
       options: { separator, decimal },
     } = useCurrency();
-    const [textValue, setTextValue] = useState(format(value ?? defaultValue));
+    const [textValue, setTextValue] = useState(
+      format(value ?? defaultValue, { minimumFractionDigits: decimalMin })
+    );
+    const isFirstRender = useRef(true);
+
+    const isInputIncomplete = useCallback(
+      (value: string) =>
+        new RegExp(
+          // Checks for a trailing `.` or a `0` (not necessarily immediately)
+          // after a `.`
+          `^\\d*\\.$|\\d*${RegexUtils.escapeChars(decimal)}\\d*0$`
```

Comment:
`decimal` is, for example, `.` or `,` right? shouldn't this be used in the first part of the regex as well? e.g. to match `1,`?